VDF.Core/ScanEngine.cs

```diff
@@ -214,35 +214,53 @@ public sealed class ScanEngine {
 					while (pauseTokenSource.IsPaused) Thread.Sleep(50);
 
 					var entry = ScanList[i];
+					float[] percent = new float[positionList.Count];
+					Dictionary<double, byte[]?>[] grayBytes = new Dictionary<double, byte[]?>[2];
+					grayBytes[0] = entry.grayBytes;
+					
+					if (Settings.CompareHorizontallyFlipped)
+					{
+						grayBytes[1] = new Dictionary<double, byte[]?>();
+						if (entry.IsImage)
+							grayBytes[1].Add(0, GrayBytesUtils.FlipGrayScale(grayBytes[0][0]!, grayScaleWidth));
+						else
+							for (var j = 0; j < positionList.Count; j++)
+							{
+								var idx = entry.GetGrayBytesIndex(positionList[j]);
+								grayBytes[1].Add(idx, GrayBytesUtils.FlipGrayScale(grayBytes[0][idx]!, grayScaleWidth));
+							}
+					}
+
 					for (var n = i + 1; n < ScanList.Count; n++) {
 						var compItem = ScanList[n];
-						if (entry.IsImage && !compItem.IsImage) continue;
-						var duplicateCounter = 0;
-						float[] percent;
-						if (entry.IsImage) {
-							percent = new float[1];
-							percent[0] = GrayBytesUtils.PercentageDifference(entry.grayBytes[0]!, compItem.grayBytes[0]!);
-							if (percent[0] < percentageDifference)
-								duplicateCounter++;
-						}
-						else {
-							percent = new float[positionList.Count];
-							for (var j = 0; j < positionList.Count; j++) {
-								percent[j] =
-									GrayBytesUtils.PercentageDifference(entry.grayBytes[entry.GetGrayBytesIndex(positionList[j])]!, compItem.grayBytes[compItem.GetGrayBytesIndex(positionList[j])]!);
-								if (percent[j] < percentageDifference) {
-									duplicateCounter++;
+						if (entry.IsImage && !compItem.IsImage) 
+							continue;
+						var isDuplicate = true;
+						int flip = 0;
+						do {
+							if (entry.IsImage) {
+								percent[0] = GrayBytesUtils.PercentageDifference(grayBytes[flip][0]!, compItem.grayBytes[0]!);
+								isDuplicate = (percent[0] < percentageDifference);
+							}
+							else {
+								for (var j = 0; j < positionList.Count; j++) {
+									percent[j] =
+										GrayBytesUtils.PercentageDifference(grayBytes[flip][entry.GetGrayBytesIndex(positionList[j])]!, compItem.grayBytes[compItem.GetGrayBytesIndex(positionList[j])]!);
+									if (!(percent[j] < percentageDifference)) {
```

Comment:
Just invert, the way its written its hard to read

Reply:
Of course I can invert it if you like, but it was intentional.
Because some lines above: `isDuplicate = (percent[0] < percentageDifference);`, so it says: set to true, if smaller.
So I prefered to keep the condition with the smaller but negate it, because isDuplicate is set to false.

_But that is probably a matter of taste. I only wrote it, because initially I inverted the condition, so used ">=" but then thought it's easier to understand the other way around._